test/WebJobs.Script.Tests/FileProvisioning/PowerShellFileProvisionerTests.cs

```diff
@@ -2,41 +2,135 @@
             await powerShellFileProvisioner.ProvisionFiles(_scriptRootPath);
-            File.Exists(Path.Combine(_scriptRootPath, "requirements.psd1"));
-            File.Exists(Path.Combine(_scriptRootPath, "profile.ps1"));
+
+            File.Exists(Path.Combine(_scriptRootPath, RequirementsPsd1FileName));
+            File.Exists(Path.Combine(_scriptRootPath, ProfilePs1FileName));
+
+            string requirementsContent = File.ReadAllText(Path.Combine(_scriptRootPath, RequirementsPsd1FileName));
+
+            string expectedContent = @"# This file enables modules to be automatically managed by the Functions service.
+# Only the Azure Az module is supported in public preview.
+# See https://aka.ms/functionsmanageddependency for additional information.
+#
+@{
+    # For latest supported version, go to 'https://www.powershellgallery.com/packages/Az'. Uncomment the next line and replace the MAJOR_VERSION, e.g., 'Az' = '2.*'
+    # 'Az' = 'MAJOR_VERSION.*'
+}";
+            Assert.Equal(requirementsContent, expectedContent, StringComparer.OrdinalIgnoreCase);
+
+            ValidateLogs(_logger, unableToReachPSGallery: true);
         }
 
         [Theory]
         [InlineData(null)]
         [InlineData("")]
         public async Task AugmentFiles_EmptyScriptRootPath_Test(string scriptRootPath)
         {
-            var powerShellFileProvisioner = new PowerShellFileProvisioner();
+            var powerShellFileProvisioner = new PowerShellFileProvisioner(_logger);
             Exception ex = await Assert.ThrowsAsync<ArgumentException>(async () => await powerShellFileProvisioner.ProvisionFiles(scriptRootPath));
             Assert.True(ex is ArgumentException);
         }
+
+        private void EnsurePowerShellFilesDoNotExist(string functionAppRootPath)
+        {
+            File.Delete(Path.Combine(functionAppRootPath, RequirementsPsd1FileName));
+            File.Delete(Path.Combine(functionAppRootPath, ProfilePs1FileName));
+        }
+
+        private void ValidateLogs(TestLogger logger, bool unableToReachPSGallery = false)
+        {
+            List<string> expectedLogs = new List<string>();
+            expectedLogs.Add($"Creating {RequirementsPsd1FileName}.");
+            expectedLogs.Add($"{RequirementsPsd1FileName} created sucessfully.");
+            expectedLogs.Add($"Creating {ProfilePs1FileName}.");
+            expectedLogs.Add($"{ProfilePs1FileName} created sucessfully.");
+
+            if (unableToReachPSGallery)
+            {
+                expectedLogs.Add("Failed to get Az module version. Edit the requirements.psd1 file when the powershellgallery.com is accessible.");
+            }
+
+            var logs = logger.GetLogMessages();
+            foreach (string log in expectedLogs)
+            {
+                Assert.True(logs.Any(l => string.Equals(l.FormattedMessage, log)));
+            }
+        }
+    }
+
+    internal class PowerShellFileProvisionerMocksPSGalleryCalls : PowerShellFileProvisioner
```

Comment:
Rename PowerShellFileProvisionerMocksPSGalleryCalls to TestPowerShellFileProvisioner and move it to its own file

Reply:
Done.